Mathlib/Algebra/Order/Group/Basic.lean

```diff
@@ -27,13 +27,22 @@ variable [OrderedCommGroup α] {m n : ℤ} {a b : α}
   rintro rfl
   simp at hn
 
-@[to_additive zsmul_strictMono_left]
+@[to_additive zsmul_left_strictMono]
 lemma zpow_right_strictMono (ha : 1 < a) : StrictMono fun n : ℤ ↦ a ^ n := fun m n h ↦
   calc
     a ^ m = a ^ m * 1 := (mul_one _).symm
     _ < a ^ m * a ^ (n - m) := mul_lt_mul_left' (one_lt_zpow' ha <| Int.sub_pos_of_lt h) _
     _ = a ^ n := by simp [← zpow_add, m.add_comm]
 
+@[deprecated (since := "2024-09-19")] alias zsmul_strictMono_left := zsmul_left_strictMono
+
+@[to_additive zsmul_left_strictAnti]
+lemma zpow_right_strictAnti (ha : a < 1) : StrictAnti fun n : ℤ ↦ a ^ n := fun m n h ↦ calc
+  a ^ n < a ^ n * a⁻¹ ^ (n - m) :=
+    lt_mul_of_one_lt_right' _ <| one_lt_zpow' (one_lt_inv'.2 ha) <| Int.sub_pos.2 h
+  _ = a ^ m := by
+    rw [inv_zpow', Int.neg_sub, ← zpow_add, Int.add_comm, Int.sub_add_cancel]
+
 @[to_additive zsmul_left_inj]
 lemma zpow_right_inj (ha : 1 < a) {m n : ℤ} : a ^ m = a ^ n ↔ m = n :=
```

Comment:
Now that you proved the above lemma, can you generalise this to
```suggestion
 lemma zpow_right_inj (ha : a != 1) {m n : ℤ} : a ^ m = a ^ n ↔ m = n :=
```

Reply:
No, because we don't assume `LinearOrder` here.

Reply:
Do you have a use case for the non-linear order version?

Reply:
No, but I prefer to postpone this until I [introduce `NoRootsOfUnity`](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Multiplicative.20version.20of.20NoZeroSMulDivisors.3F), because this is the right typeclass assumption for the lemma with `ne` assumption.